### PR TITLE
`Development`: Avoid unnecessary REST calls in course details

### PR DESCRIPTION
**Pull request by rriyaldhi**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The learning goals, tutorial groups, and prerequisites of a course are fetched already when calling courses/{courseId}/for-dashboard API. Therefore separate calls are not needed when visiting a course. The API calls are shown in this figure:

before

### Description

1. Remove /courses/{courseId}/for-dashboard API call from MetisConversationService
2. Always call /courses/{courseId}/for-dashboard API from the Course Overview Component and pass the fetched course to MetisConversationService
3. Remove additional API calls to get Learning Goals, Prerequisites and Tutorial Groups

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Course with Learning Goals, Prerequisites, and Tutorial Groups

1. Log in to Artemis as a Student
4. Navigate to Course
5. Open network tabs in browser
6. Select 1 of the Course
7. Verify that there are no additional API calls to retrieve learning goals, prerequisites, and tutorial groups
8. Verify that learning goals, prerequisites and tutorial groups are accessible in the course detail

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-overview.component.ts | 75.6% | 85.49% |
| metis-conversation.service.ts | 71.42% | 85% |

### Screenshots

After the changes in this PR, no separate API calls are made:
image

Learning Goals and Prerequisites:
image

Tutorial Groups:
image

### CODE REVIEW

**Review by TheZoker — APPROVED** [review]
Tested during testing session on TS2 and looks good to me!

**Review by jakubriegel — APPROVED** [review]
Code looks fine

File: src/main/webapp/app/shared/metis/metis-conversation.service.ts [hunk 1]
```diff
@@ -184,20 +184,10 @@ export class MetisConversationService implements OnDestroy {
...
-            map((res: HttpResponse<Course>) => {
-                return res.body;
-            }),
-            switchMap((course: Course) => {
-                this._course = course;
-                return this.conversationService.getConversationsOfUser(this._course.id ?? 0);
-            }),
+    public setUpConversationService = (course: Course): Observable<never> => {
+        this._courseId = course.id!;
+        this._course = course;
```
Comment (pellumbbaboci) [inline src/main/webapp/app/shared/metis/metis-conversation.service.ts:189]: I think you need to have `this.setIsLoading(true);` here. Or is this removal intended?

Comment (rriyaldhi) [reply src/main/webapp/app/shared/metis/metis-conversation.service.ts:189]: It was intentional since the api call is removed here, so loading is not needed.
